pkg/R/RDD.R

```diff
@@ -62,45 +76,65 @@ setMethod("initialize", "PipelinedRDD", function(.Object, prev, func, jrdd_val)
     # This transformation is the first in its stage:
     .Object@func <- func
     .Object@prev_jrdd <- getJRDD(prev)
-    # Since this is the first step in the pipeline, the prev_serialized
-    # is same as serialized here.
-    .Object@env$prev_serialized <- .Object@env$serialized
+    .Object@env$prev_serializedMode <- prev@env$serializedMode
+    # NOTE: We use prev_serializedMode to track the serialization mode of prev_JRDD
+    # prev_serializedMode is used during the delayed computation of JRDD in getJRDD
+
+    .Object@colNames <- prev@colNames
```

Comment:
This is not correct, `func` can do anything (including change the number columns, or column name), so PipelinedRDD@colNames should be `list()`.

Only in few of cases, `.Object@colNames` can be `prev@colNames`, for example, reserialization.